presto-cli/src/main/java/com/facebook/presto/cli/Console.java

```diff
@@ -374,17 +375,12 @@ private static MemoryHistory getHistory()
     @VisibleForTesting
     static MemoryHistory getHistory(File historyFile)
     {
-        if (!historyFile.canWrite() || !historyFile.canRead()) {
-            System.err.printf("WARNING: History file is not readable/writable: %s. " +
-                            "History will not be available during this session.%n",
-                    historyFile.getAbsolutePath());
-            MemoryHistory history = new MemoryHistory();
-            history.setAutoTrim(true);
-            return history;
-        }
-
         MemoryHistory history;
         try {
+            //  try creating the history file and its parents to check
+            // whether the directory tree is readable/writeable
+            createParentDirs(historyFile.getParentFile());
```

Comment:
it should be CLI responsibility to create missing dirs (and missing dirs may be result of misconfig).
Do other tools work like that?

Reply:
Other clients may be all over the place in terms of behavior, but I think it makes sense for the presto-cli to do it.
Previously, `FileHistory` was already doing the exact same thing, but on flush, which is late if the parent dirs are not readable/writable, etc. Instead we do it here on init.